tests/test_reactive_kubernetes_worker.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from unittest import mock
+
+from reactive.kubernetes_worker import open_close_ports_if_needed
+
+def patch_fixture(patch_target):
+    @pytest.fixture()
+    def _fixture():
+        with mock.patch(patch_target) as m:
+            yield m
+    return _fixture
+
+
+mock_close_port   = patch_fixture('charmhelpers.core.hookenv.close_port')
+mock_open_port    = patch_fixture('charmhelpers.core.hookenv.open_port')
+mock_config_get   = patch_fixture('charmhelpers.core.hookenv.config')
+mock_opened_ports = patch_fixture('charmhelpers.core.hookenv.opened_ports')
+mock_log          = patch_fixture('charmhelpers.core.hookenv.log')
+
+def test_open_close_ports_dup_in_config(mock_close_port,
+                                        mock_open_port,
+                                        mock_config_get,
+                                        mock_opened_ports,
+                                        mock_log):
+    # Test dups on config + real icmp entry on opened_ports
+    mock_opened_ports.return_value = ["80/tcp", "80/udp", "443/tcp","icmp"]
+    # Check if spaces are processed correctly
+    mock_config_get.return_value = {
+        "open-ports": "80, 80,443",
+    }
+    # Should not call 80 but close 443
```

Comment:
won't it bounce all the ports? if you call it with 80/tcp twice in a row, I would expect the second call would get close_port(80) and then open_port(80) called. Is that correct?

Reply:
Because open/close-port is more complex than that. We can have for example:
old_config: 80-90/tcp
new_config: 80/udp, 82-90/tcp

The quickest (and dirtiest) way is to close everything and then open corresponding ports. Also, Juju is ignoring when I request: close-port 80-81/tcp if I have 80-90 range (probably a bug we will have to fill later). 

Therefore, I would have to open all ranges  into a dict with list for tcp, udp and a bool for icmp. I have an ongoing change for that but it is way more complex.

I can make this change, but then the code will be more difficult to grasp.